sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala

```diff
@@ -391,6 +390,22 @@ private[parquet] class ParquetRowConverter(
             updater.setLong(timestampRebaseFunc(value))
           }
         }
+      case t: DayTimeIntervalType =>
```

Comment:
Hm, I don't think `DayTimeIntervalType` is matched with Parquet's `TIME` type (https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#time) because `DayTimeIntervalType` contains days? For the similar reasons, we don't support unsigned types too.

Reply:
+1, mapping day-time interval to parquet time type is incorrect

Reply:
Thank you for your review.

I understand that this issue can be fixed by restringing the DayTimeIntervalType to cases where the interval of fields is from hour to second.

Reply:
Pushed a change that addresses your request.